library/cartridge_connect_to_membership.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+
+import math
+import pkgutil
+
+if pkgutil.find_loader('ansible.module_utils.helpers'):
+    import ansible.module_utils.helpers as helpers
+else:
+    import module_utils.helpers as helpers
+
+argument_spec = {
+    'console_sock': {'required': True, 'type': 'str'},
+    'hostvars': {'required': True, 'type': 'dict'},
+    'play_hosts': {'required': True, 'type': 'list'},
+}
+
+BATCH_SIZE = 32
+
+
+def connect_instances(control_console, play_hosts, instance_names, advertise_uris):
+    connect_errors, _ = control_console.eval_res_err('''
+        local advertise_uris = ...
+
+        local cartridge = require('cartridge')
+        local fiber = require('fiber')
+
+        local count = 0
+        local channel = fiber.channel()
+
+        for _, advertise_uri in ipairs(advertise_uris) do
+            count = count + 1
+
+            fiber.create(function()
+                local _, err = cartridge.admin_probe_server(advertise_uri)
+                channel:put({
+                    advertise_uri = advertise_uri,
+                    err = err and err.err or err,
+                })
+            end)
+        end
+
+        local errors = {}
+        for _ = 1, count do
+            local res = channel:get(60)
+            if res == nil then
+                table.insert(errors, {msg = 'Failed to get probe result: timeout'})
+                break
+            end
+            if res.err ~= nil then
+                table.insert(errors, {
+                    msg = ("Failed to probe server '%%s': %s"):format(res.err),
+                    advertise_uri = res.advertise_uri,
+                })
+            end
+        end
+        return errors
+    ''', advertise_uris)
+
+    errors = []
+    warnings = []
+
+    for error in connect_errors:
+        if not error['advertise_uri']:
+            errors.append(error['msg'])
+            break
+
+        instance_name = instance_names[error['advertise_uri']]
+        msg = error['msg'] % instance_name
+
+        if instance_name in play_hosts:
+            errors.append(msg)
+        else:
+            warnings.append(msg)
```

Comment:
I'm not sure that collecting such warinings is good.
Imagine that you have 500 instances configured and start to start them with limits by 5 instances each time.
You'll have 495 warnings.